ouroboros-consensus/test-storage/Test/Ouroboros/Storage/ImmutableDB/StateMachine.hs

```diff
@@ -233,6 +236,15 @@ run ImmutableDBEnv { db, internal, registry, varNextId } runCorruption its cmd =
       closeDB db
       reopen db valPol
       Tip <$> getTip db
+    ReopenInThePast valPol curSlot -> do
+      mapM_ iteratorClose (unWithEq <$> its)
+      closeDB db
+      atomically $ writeTVar varCurSlot curSlot
+      reopen db valPol
+      -- Reset the current slot back to a value that won't get us into trouble
+      -- when we try to reopen the database
+      atomically $ writeTVar varCurSlot maxBound
```

Comment:
Use `bracket` for temporarily changing the `varCurSlot` (in case `reopen` throws an exception).